src/Strategy/Go/GoList.hs

```diff
@@ -5,36 +5,72 @@ module Strategy.Go.GoList (
   Require (..),
 ) where
 
-import Control.Effect.Diagnostics
-import Data.ByteString.Lazy qualified as BL
+import Control.Effect.Diagnostics hiding (fromMaybe)
+import Data.Aeson (FromJSON, withObject, (.!=), (.:), (.:?))
+import Data.Aeson.Internal (formatError)
+import Data.Aeson.Types (parseJSON)
 import Data.Foldable (traverse_)
-import Data.Maybe (mapMaybe)
-import Data.String.Conversion (decodeUtf8)
+import Data.Maybe (fromMaybe)
+import Data.String.Conversion (toText)
 import Data.Text (Text)
-import Data.Text qualified as Text
-import DepTypes
-import Effect.Exec
-import Effect.Grapher
+import DepTypes (Dependency)
+import Effect.Exec (
+  AllowErr (Never),
+  Command (..),
+  Exec,
+  ExecErr (CommandParseError),
+  execThrow,
+ )
+import Effect.Grapher (deep, direct, label)
 import Graphing (Graphing)
 import Path
-import Strategy.Go.Transitive (fillInTransitive)
-import Strategy.Go.Types
+import Strategy.Go.Transitive (decodeMany, fillInTransitive)
+import Strategy.Go.Types (
+  GolangGrapher,
+  graphingGolang,
+  mkGolangPackage,
+  mkGolangVersion,
+ )
 import Types (GraphBreadth (..))
 
 data Require = Require
   { reqPackage :: Text
   , reqVersion :: Text
+  , isDirect :: Bool
   }
   deriving (Eq, Ord, Show)
 
-golistCmd :: Command
-golistCmd =
+goListJsonCmd :: Command
+goListJsonCmd =
   Command
     { cmdName = "go"
-    , cmdArgs = ["list", "-m", "all"]
+    , cmdArgs = ["list", "-m", "-json", "all"]
     , cmdAllowErr = Never
     }
+data GoListModule = GoListModule
+  { path :: Text
+  , version :: Maybe Text
+  , isMain :: Bool
+  , isIndirect :: Bool
+  }
+  deriving (Show, Eq, Ord)
+
+instance FromJSON GoListModule where
+  parseJSON = withObject "GoListModule" $ \obj ->
+    GoListModule <$> obj .: "Path"
+      <*> obj .:? "Version"
+      <*> (obj .:? "Main" .!= False)
+      <*> (obj .:? "Indirect" .!= False)
+
+toRequires :: [GoListModule] -> [Require]
```

Comment:
Is there a reason you moved `toRequires` out of the function block when we aren't using it anywhere else?